src/app/encoder.cpp

```diff
@@ -174,35 +173,30 @@ uint16_t encodeReadOnOffCommand(uint8_t * buffer, uint16_t buf_length, uint8_t d
     return result;
 }
 
-uint16_t _encodeIdentifyClusterCommand(uint8_t * buffer, uint16_t buf_length, uint8_t command_id, uint8_t destination_endpoint,
-                                       uint16_t * identify_duration)
-{
+/*
+ * Identify Cluster commands
+ *
+ * Pick cluster id as 0x0003 for now
+ */
 
+uint16_t encodeIdentifyCommand(uint8_t * buffer, uint16_t buf_length, uint8_t destination_endpoint, uint16_t duration)
+{
     BufBound buf    = BufBound(buffer, buf_length);
-    uint16_t result = _encodeClusterSpecificCommand(buf, destination_endpoint, 0x0003, command_id);
-
-    if (identify_duration)
-    {
-        buf.PutLE16(*identify_duration);
-    }
+    uint16_t result = _encodeClusterSpecificCommand(buf, destination_endpoint, 0x0003, 0x00);
+    CHECK_COMMAND_LENGTH(result, "Identify");
 
-    result = buf.Fit() ? buf.Written() : 0;
+    buf.PutLE16(duration);
 
+    CHECK_COMMAND_LENGTH((result = buf.Fit() ? buf.Written() : 0), "Identify");
```

Comment:
Again, I am unconvinced by this "assign in a condition" pattern.

Reply:
Reverted.